internal/provider/azure/environ.go

```diff
@@ -233,19 +233,59 @@ func (env *azureEnviron) initEnviron(ctx context.Context) error {
 // PrepareForBootstrap is part of the Environ interface.
 func (env *azureEnviron) PrepareForBootstrap(ctx environs.BootstrapContext, _ string) error {
 	if ctx.ShouldVerifyCredentials() {
-		if err := verifyCredentials(env, ctx); err != nil {
+		if err := verifyCredentials(ctx, env); err != nil {
 			return errors.Trace(err)
 		}
 	}
 	return nil
 }
 
-// Create is part of the Environ interface.
-func (env *azureEnviron) Create(ctx envcontext.ProviderCallContext, args environs.CreateParams) error {
-	if err := verifyCredentials(env, ctx); err != nil {
+// ValidateProviderForNewModel is part of the [environs.ModelResources] interface.
+func (env *azureEnviron) ValidateProviderForNewModel(ctx context.Context) error {
```

Comment:
Hi Ian, im slightly confused for this validation method, just wanted to clarify with you if these were intended.
1. Does it passes the validation check if resourceGroupName is empty?
2. Are env.config.resourceGroupName and env.resourceGroup the same thing? If so, should we choose one for validation?

Reply:
`env.config.resourceGroupName` and `env.resourceGroupName` are different. The first one is what could be specified by the user when adding a model using the `resource-group` model config attribute. However, the actual resource group may also come from a managed identity. So there's a subtle difference and the existing unit tests are supposed to cover those cases.

Reply:
Thanks for the clarifications Ian :)